lib/pbkit/pbkit.c

```diff
@@ -2260,6 +2260,7 @@ void pb_end(DWORD *pEnd)
 #endif
 
 #ifdef DBG
+    if (pEnd!=pb_PushNext) debugPrint("pb_end: input pointer invalid or not following previous write addresses\n");
```

Comment:
We should stop and scream (`assert`). Using the `debugPrint` macro can lead to missing these important error messages. Won't block merge for this but it should be changed

Reply:
Agreed.

We couldn't do this before this PR because any use of `pb_push` would automatically trigger these.

I had also [commented on Discord yesterday](https://discordapp.com/channels/428359196719972353/428360618102226946/628372619330060303):

> I have tried #158 in `DBG` mode (using XQEMU) and triangle sample seems to work, but mesh sample seems to generate warnings - however, at least some of them also happen in master.
> In particular, during startup, both samples cause "pb_push_to: new write address invalid or not following previous write addresses".
>
> [Note: I think the following comments are about the mesh sample - not sure if triangle is also affected]
> - master also causes it during runtime (and then causes XQEMU to assert due CPU / GPU both writing at the same time)
> - #158 continues but also causes "pb_push_to: begin-end block musn't exceed 128 dwords"

Since then, the last issue should have been fixed in #183 (Issue #178).

I had forgotten about the other ones, but my Discord messages imply those messages still trigger.
We should probably look into this before a merge.
I believe this PR should have fixed those messages.

Reply:
Fixed by adding assertions in dcb7a59 and bae0bc8 respectively. :+1:

Reply:
This will definitely need testing with all nxdk samples now - I assume they'll hit the assertion in `DBG` mode now (unless the bugs were caused by the `p+17` matrix return value).

Reply:
I just rebuilt the `mesh`, `sdl` and `triangle` samples using `make clean && make DBG=y DEBUG=y` and tested them in XQEMU, works the same as without the flags.

Reply:
Those issues have been fixed by now (#187 and #183, possibly more).